presto-hive/src/main/java/com/facebook/presto/hive/metastore/HivePrivilegeInfo.java

```diff
@@ -103,6 +104,23 @@ public boolean isContainedIn(HivePrivilegeInfo hivePrivilegeInfo)
                         (!isGrantOption() && hivePrivilegeInfo.isGrantOption())));
     }
 
+    public PrivilegeInfo toPrivilegeInfo()
```

Comment:
You can keep HivePrivilegeInfo, just use it as a helper to create PrivilegeInfo objects. So you don't have to do so much copying/translation; this isn't something that would have a lot of GC overhead, but in general that's something to watch out for.